pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -464,27 +465,46 @@ func applyUpstreamTLSSettings(cluster *v2.Cluster, tls *networking.TLSSettings)
 				cluster.Name)
 			return
 		}
+
 		cluster.TlsContext = &auth.UpstreamTlsContext{
-			CommonTlsContext: &auth.CommonTlsContext{
-				TlsCertificates: []*auth.TlsCertificate{
-					{
-						CertificateChain: &core.DataSource{
-							Specifier: &core.DataSource_Filename{
-								Filename: tls.ClientCertificate,
-							},
+			CommonTlsContext: &auth.CommonTlsContext{},
+			Sni:              tls.Sni,
+		}
+
+		// Fallback to file mount secret instead of SDS if meshConfig.sdsUdsPath isn't set or tls.mode is TLSSettings_MUTUAL.
```

Comment:
add logs should be good; I think sds/file-mount have to be co-exist

Reply:
They can co-exist, but my point is we should only have ONE flag to determine which mode should we use (i.e., sds or file mount), if sds flag is on, sdsUdsPath being empty is an error and we should report, instead of fallback, right?

Reply:
May I ask why we cannot use sds when tls.mode is TLSSettings_MUTUAL?

Reply:
because ```TLSSettings_MUTUAL``` is for user bring their own key/cert

Reply:
@wattli I agree, the check sounds should be in some earlier called funcs before hitting cluster.go (like validate sdsudspath in ValidateMeshConfig),will add validation later after we have basic e2e works.

Reply:
can we supply user's certs as well through our SDS?

Reply:
probably in the near future(we do see customer request to distribute their cert through sds), right now want want to make basic scenario works e2e first.